### PR TITLE
Remove short_name method

### PR DESCRIPTION
**Pull request by dstrain115**
- Replace with wire_symbol(reg=None)

For item 2 in #728

### CODE REVIEW

File: qualtran/_infra/controlled.py [hunk 1]
```diff
@@ -433,11 +433,13 @@ def _unitary_(self):
...
-    def wire_symbol(self, reg: Register, idx: Tuple[int, ...] = tuple()) -> 'WireSymbol':
+    def wire_symbol(self, reg: Optional[Register], idx: Tuple[int, ...] = tuple()) -> 'WireSymbol':
+        from qualtran.drawing import Text
+
+        if reg is None:
+            return Text(f'C[{self.subbloq.wire_symbol(reg=None)}]')
         if reg.name not in self.ctrl_reg_names:
             # Delegate to subbloq
-            print(self.subbloq)
-            print(type(self.subbloq))
```
Comment (mpharrigan) [inline qualtran/_infra/controlled.py:440]: whoops

**Comment by mpharrigan** [comment]
One of the motivations was that the author of the `wire_symbol` method could declare that "the symbols are sufficient to tell what bloq this is, I don't need an additional label in the diagrams". For example: writing "CNOT" over a circle and modplus in the musical score diagram just adds clutter. Follow on work would aim to try to make some of the symbols more expressive. For example: the Add bloq could have its symbols be `x` and `x+y` and then get rid of the label.

Could you add support for this use case? both in the types and docs and in the musical score drawer?

**Comment by dstrain115** [comment]
> One of the motivations was that the author of the `wire_symbol` method could declare that "the symbols are sufficient to tell what bloq this is, I don't need an additional label in the diagrams". For example: writing "CNOT" over a circle and modplus in the musical score diagram just adds clutter. Follow on work would aim to try to make some of the symbols more expressive. For example: the Add bloq could have its symbols be `x` and `x+y` and then get rid of the label.
> 
> Could you add support for this use case? both in the types and docs and in the musical score drawer?

Would returning `Text('')` be sufficient here?  (wire_symbol() should return a WireSymbol)

I will probably leave modifying the symbols of specific gates to a different PR (also, will only do obvious ones, like arithmetic operators, since I am not super familiar with all of these bloqs)

**Comment by mpharrigan** [comment]
I sortof defer to you on how to signal "no label", but whatever it is: it should be documented and the `musical_score` drawer should avoid drawing the label

**Comment by dstrain115** [comment]
@mpharrigan I think this is ready for review.  Please let me know if I missed anything.

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**